File: python/src/wait_tx_status.py
import time
import requests
from typing import Literal

# トランザクションハッシュを指定してトランザクションの状態を確認する関数
async def wait_tx_status(
  hash: str,
  node_url: str,
  transaction_status: Literal[
    "confirmed", "unconfirmed", "partial"
  ],
) -> None:
  print(f"{transaction_status}状態まで待機中..")
  for _ in range(100):
    time.sleep(1)
    # トランザクションハッシュからステータスを確認
    status = requests.get(
      f"{node_url}/transactionStatus/{hash}",
      headers={"Content-Type": "application/json"},
    ).json()
    # 指定したトランザクションステータスになっていたら結果を表示させる
    if status["code"] == "ResourceNotFound":
      continue
    elif status["group"] == transaction_status:
      print(f"{status['group']}完了!")
      print("承認結果", status["code"])
      print("承認状態", status["group"])
      print("トランザクションハッシュ", hash)
      print("ブロック高", status["height"])
      print("Symbolエクスプローラー ")
      print(f"https://testnet.symbol.fyi/transactions/{hash}")
      return
    elif status["group"] == "failed":
      print("承認結果:", status["code"])
      return

  raise Exception("トランザクションが確認されませんでした。")

File: typescript/src/waitTxStatus.ts
//トランザクションハッシュを指定してトランザクションの状態を確認する関数
export async function waitTxStatus(
  hash: string,
  nodeUrl: string,
  transactionStatus: "confirmed" | "unconfirmed" | "partial",
) {
  console.log(`${transactionStatus}状態まで待機中..`)
  await new Promise(async (resolve, reject) => {
    for (let i = 0; i < 100; i++) {
      await new Promise((res) => setTimeout(res, 1000))
      //トランザクションハッシュからステータスを確認
      const status = await fetch(
        new URL("/transactionStatus/" + hash, nodeUrl),
        {
          method: "GET",
          headers: { "Content-Type": "application/json" },
        },
      ).then((res) => res.json())
      //指定したトランザクションステータスになっていたら結果を表示させる
      if (status.code === "ResourceNotFound") {
        continue
      } else if (status.group === transactionStatus) {
        console.log(`${status.group}完了!`)        
        console.log("承認結果",status.code)
        console.log("承認状態",status.group)
        console.log("トランザクションハッシュ",status.hash)
        console.log("ブロック高",status.height)
        console.log("Symbolエクスプローラー ",
          `https://testnet.symbol.fyi/transactions/${hash}`,
        )
        resolve({})
        return
      } else if (status.group === "failed") {
        console.log("承認結果 ", status.code)
        resolve({})
        return
      }
    }
    reject(new Error("トランザクションが確認されませんでした。"))
  })
}
